tcf_website/static/reviews/review_form.js

```diff
@@ -0,0 +1,136 @@
+/*
+ * Cascading dropdown data population for the "new review" form
+ * in review.py and reviews/new.html
+ * Author: j-alicia-long, 11/22/2020
+*/
+
+// Executed when DOM is ready
+jQuery(function($) {
+    // Clear & disable sequenced dropdowns
+    clearDropdown("#subject");
+    clearDropdown("#courseID");
+    clearDropdown("#instructor");
+    clearDropdown("#semester");
+    $("#courseName").val("");
+    // Enable subject selector, disable the following
+    $("#subject").prop("disabled", false);
+    $("#courseID").prop("disabled", true);
+    $("#instructor").prop("disabled", true);
+    $("#semester").prop("disabled", true);
+
+    // Fetch all subdepartment data from API
+    var subdeptEndpoint = "/api/subdepartments/";
+    $.getJSON(subdeptEndpoint, function(data) {
+        // Sort departments alphabetically by mnemonic
+        data.sort(function(a, b) {
+            return a.mnemonic.localeCompare(b.mnemonic);
+        });
+
+        // Generate option tags
+        $.each(data, function(i, subdept) {
+            $("<option />", {
+                val: subdept.id,
+                text: subdept.mnemonic
+            }).appendTo("#subject");
+        });
+        return this;
+    });
+
+    // Fetch course data on subject select
+    $("#subject").change(function() {
+        // Clear & disable sequenced dropdowns
+        clearDropdown("#courseID");
+        clearDropdown("#instructor");
+        clearDropdown("#semester");
+        $("#courseName").val("");
+        // Enable course selector, disable the following
+        $("#courseID").prop("disabled", false);
+        $("#instructor").prop("disabled", true);
+        $("#semester").prop("disabled", true);
+
+        // Fetch course data from API, based on selected subdepartment
+        var subdeptID = $("#subject").val();
+        var pageSize = "1000";
+        var courseEndpoint = "/api/courses/?subdepartment=" + subdeptID +
```

Comment:
Would it look better if we use [`$.param()`](https://api.jquery.com/jQuery.param/) or [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals)?

Reply:
oh I forgot about JS template literals, good point